src/main/java/org/datadog/jenkins/plugins/datadog/util/TagsUtil.java

```diff
@@ -0,0 +1,43 @@
+package org.datadog.jenkins.plugins.datadog.util;
+
+import net.sf.json.JSONArray;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class TagsUtil {
+
+    public static Map<String, Set<String>> merge(Map<String, Set<String>> dest, Map<String, Set<String>> orig) {
+        if (dest == null) {
+            dest = new HashMap<>();
+        }
+        if (orig == null) {
+            orig = new HashMap<>();
+        }
+        for (String oName: orig.keySet()){
+            Set<String> dValues = dest.containsKey(oName) ? dest.get(oName) : new HashSet<String>();
+            if (dValues == null) {
+                dValues = new HashSet<>();
+            }
+            Set<String> oValues = orig.get(oName);
+            if (oValues != null) {
+                dValues.addAll(oValues);
+            }
+            dest.put(oName, dValues);
+        }
+        return dest;
+    }
+
+    public static JSONArray convertTagsToJSONArray(Map<String, Set<String>> tags){
+        JSONArray result = new JSONArray();
+        for (String name : tags.keySet()) {
+            Set<String> values = tags.get(name);
+            for (String value : values){
+                result.add(String.format("%s:%s", name, value));
```

Comment:
Let's deal with the case where a tag is just a key and has no value

Reply:
Will add this as part of another PR. See my previous comment